scripts/ui/mode_selection.py

```diff
@@ -0,0 +1,170 @@
+import bpy
+
+class ARF_OT_Armature_Mode(bpy.types.Operator):
+    bl_label = "Armature Mode"
+    bl_idname = "arf.armature_mode"
+
+    def execute(self, context):
+        for obj in context.scene.objects:
+            if (obj.type == "ARMATURE"):
+                obj.hide_set(False)
+            else:
+                obj.hide_set(True)
+        return {'FINISHED'}
+
+
+class ARF_OT_Skeleton_Mode(bpy.types.Operator):
+    bl_label = "Skeleton Mode"
+    bl_idname = "arf.skeleton_mode"
+
+    def execute(self, context):
+        for obj in context.scene.objects:
+            if (obj.ARF.object_type == "BONE"):
+                obj.hide_set(False)
+            elif (obj.ARF.object_type != "NONE"):
+                obj.hide_set(True)
+        return {'FINISHED'}
+
+
+class ARF_OT_Smart_Object_Mode(bpy.types.Operator):
+    bl_label = "Smart Object Mode"
+    bl_idname = "arf.smart_object_mode"
+
+    def execute(self, context):
+        # First check if we selected at least one object.
+        # To not hide all objects by mistake.
+        if context.selected_objects == []:
+            return {'FINISHED'}
+
+        selected = context.selected_objects
+        # Then, hide all objects
+        for obj in context.scene.objects:
+            obj.hide_set(True)
+
+        # Now get all objects tied to geom nodes tied to the selected object to unhide those
+        for selection in selected:
+            for modifier in selection.modifiers:
+                if modifier.type != 'NODES':
+                    continue
+                # Disabled geom node modifiers sometimes do not contain a node_group
+                if (modifier.node_group == None):
+                    continue
+
+                for input in modifier.node_group.inputs:
+                    if not (isinstance(input, bpy.types.NodeSocketInterfaceObject)):
+                        continue
+
+                    # For some reason input.default_value is a copy of the object and not the actual
+                    # object in the scene, so we need to get the actual object in the scene.
+                    inputobj = bpy.context.scene.objects.get(input.default_value.name)
+                    inputobj.hide_set(False)
+                    # We also need to set every parent to also be visible
+                    parent = inputobj.parent
+                    while (parent != None):
+                        parent.hide_set(False)
+                        parent = parent.parent
+
```

Comment:
Ah, I see you are looping over the modifier list. Sadly that has the effect of displaying the objects that directly modify the bones, which may not be the objects that the user can edit (instead being intermediary objects). These objects can be rather large and clutter the viewport, due to their mathematical requirement of being large, so it's best if they stay hidden. The preferred way to display the correct objects is to loop over `selection.ARF.bone_properties.geometry_constraints` and then getting `constraint_object` from that. e.g.:

```python
for selection in selected:
    for geometry_constraint in selection.ARF.bone_properties.geometry_constraints:
        geometry_constraint.constraint_object.hide_set(False)
```

Let me send you a .blend file where an object has those geometry_constraints properly filled in so that you can test the code!